### PR TITLE
Detection of cycle in a Linked List using C++

### PR DESCRIPTION
**Pull request by somya1252**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

File: C-Plus-Plus/ds/detectionOfCycle_likedList.cpp [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+SAMPLE INPUT
+7
+1 2 3 4 5 6 7
+3
+
+SAMPLE OUTPUT
+Cycle is Present
+
+cout<<endl;
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/detectionOfCycle_likedList.cpp]: Its not even commented. What is the meaning of this?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/ds/detectionOfCycle_likedList.cpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    return 0;
+}
+
+//SAMPLE INPUT
+//7
+//1 2 3 4 5 6 7
+//3
+//
+//SAMPLE OUTPUT
+//Cycle is Present
```
Comment (siddharth25pandey) [inline C-Plus-Plus/ds/detectionOfCycle_likedList.cpp]: Use Multi line comment

File: C-Plus-Plus/ds/detectionOfCycle_likedList.cpp [hunk 3]
```diff
@@ -0,0 +1,106 @@
+/*Below is the code which is used to find whether a cycle is present in a linked list.
+It uses 2 pointer slow and fast(which travels at speed twice that of slow).
+If they meet at any position while traversing the linked list then a cycle is
+present otherwise cycle is not present */
+#include <iostream>
```
Comment (siddharth25pandey) [inline C-Plus-Plus/ds/detectionOfCycle_likedList.cpp]: Use **bits/stdc++**

File: C-Plus-Plus/ds/detectionOfCycle_likedList.cpp [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        }
+    }
+    return false;
+}
+
+int main()
+{
+    node* head = NULL;
+    int n;
+    //cout<<"Enter the number of nodes:";
```
Comment (siddharth25pandey) [inline C-Plus-Plus/ds/detectionOfCycle_likedList.cpp]: Write it in print form for better UI

File: C-Plus-Plus/ds/detectionOfCycle_likedList.cpp [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    cin>>n;
+    for(int i=0;i<n;i++){
+       int x;
+       cin>>x;
+       insertAtTail(head,x);
+    }
+    int pos;
+    //cout<<"Enter the position where you want to make a cycle";
```
Comment (siddharth25pandey) [inline C-Plus-Plus/ds/detectionOfCycle_likedList.cpp]: Write it in print form for better UI

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@somya1252 Do the suggested changes

**Comment by siddharth25pandey** [comment]
@somya1252 Also add Time & Space Complexity required for doing this task

File: C-Plus-Plus/ds/detectionOfCycle_likedList.cpp (hunk 5 above)
Comment (somya1252) [reply C-Plus-Plus/ds/detectionOfCycle_likedList.cpp]: Should I uncomment it

Comment (siddharth25pandey) [reply C-Plus-Plus/ds/detectionOfCycle_likedList.cpp]: Yes

Comment (UG-SEP) [reply C-Plus-Plus/ds/detectionOfCycle_likedList.cpp]: @siddharth25pandey  and @HarshCasper  please review my PR also and there is any problem please tell me i present my PR 2 days ago but there is no reply please review #4724

File: C-Plus-Plus/ds/detectionOfCycle_likedList.cpp [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+/*
+SAMPLE INPUT
+7
+1 2 3 4 5 6 7
+3
+SAMPLE OUTPUT
+Enter the number of nodes:
+Enter the position where you want to make a cycle:
+Cycle is Present
```
Comment (siddharth25pandey) [inline C-Plus-Plus/ds/detectionOfCycle_likedList.cpp]: Sample I/O is **not matching** with execution of the Program

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]

**Comment by siddharth25pandey** [comment]
@somya1252 Not done yet [here](https://github.com/TesseractCoding/NeoAlgo/pull/4898#discussion_r606619112)

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**